jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -24,9 +24,58 @@
 from jdaviz.core.tools import ICON_DIR
 from jdaviz.core.custom_traitlets import IntHandleEmpty
 
+from scipy.interpolate import make_interp_spline
+
 __all__ = ['PlotOptions']
 
 
+class SplineStretch:
+    """
+    A class to represent spline stretches.
+
+    Attributes:
+        k (int): Degree of the smoothing spline.
+        bc_type (str or None): Boundary condition type.
+        t (array-like or None): Array of knot positions.
+        x (array-like): The x-coordinates of the data points.
+        y (array-like): The y-coordinates of the data points.
+        spline (object): Interpolating spline.
+    """
+
+    def __init__(self):
+        """
+        Initializes the SplineStretch class with default knot values.
+        """
+        self.k = 3
+        self.bc_type = None
+        self.t = None
+        self.update_knots(
+            x=np.array([0, 0.1, 0.2, 0.7, 1]),
+            y=np.array([0, 0.05, 0.3, 0.9, 1])
+        )
+
+    def __call__(self, values, out=None, clip=False):
+        # TODO: Review the 'out' and 'clip' parameters once glue upstream is updated.
+        # Depending on their implementation, this method may need adjustment.
+        return self.spline(values)
+
+    def update_knots(self, x, y):
+        if len(x) != len(y):
+            raise ValueError("x and y must be the same length.")

```

Comment:
This is not tested. Also, since this is called internally by a plugin, is it nicer to silently return as a no-op here? That way, user won't see anything changing and no ugly traceback.

Reply:
Regarding Testing: You're right. This specific case hasn't been covered in the tests. I will work on adding the necessary tests to ensure this behavior is correctly verified.

Regarding Silent Return: I initially added the ValueError to provide clear feedback about any potential misuse of the update_knots method. However, I see your point about improving the user experience, especially when this is an internal call.   

Would you also @kecnry  recommend proceeding with the silent return (no-op) approach, or do you think there's value in keeping the exception to catch errors?

Reply:
its a matter of preference - I always prefer exceptions myself 🤷‍♂️ .  Ultimately this won't be possible to trigger from the UI and we haven't decided on the API for the users to set these manually, so I think either way is fine for now and we can defer the actual decision until the API to edit the knots.

Reply:
Thank you for the feedback. This current approach is suitable given our discussion and the tests already in place. I'll keep this open for a bit longer in case there are any more thoughts, and if not, I'll proceed with the merge.

Reply:
I merged it. Thanks!